### DISCUSSION TITLE
Windows 7 No longer supported?

### DISCUSSION THREAD

**Discussion by Murosel** [mifi/lossless-cut#1476] [Support] [answered]
While 3.50.0 works fine on Windows 7, v3.52.0 does not; it reports an error of cannot find GetPackageFamilyName entry point in Kernel32.dll (works fine on Windows 10)

It seems odd to drop Windows 7 support in a minor update so I suspect this is either a problem with update, or I am missing a Windows Update at my end.

Any clues would be appreciated,

Thanks

**Comment by mifi** [comment, marked as answer]
Hi. Unfortunately this is probably because I upgraded Electron to latest version 23 (from v22). Electron 23 uses chromium and chromium has decided to drop support for Windows 7, 8, 8.1.

More info:
- https://www.electronjs.org/docs/latest/breaking-changes#removed-windows-7--8--81-support
- https://support.google.com/chrome/thread/185534985/sunsetting-support-for-windows-7-8-8-1-in-early-2023?hl=en
- https://www.electronjs.org/blog/windows-7-to-8-1-deprecation-notice

I could delay updating to Electron 23, but Electron 22 reaches the end of its support cycle on **May 30 2023** and I don't want to stay on an unsupported version, so that would just be slightly delaying the inevitable. Note: electron they might extend support for electron 22 past may 2023: https://github.com/electron/electron/issues/36909 but I wouldn't bet on it.

btw:
- [Microsoft ended support for Windows 7 on January 14, 2020](https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962)
- [Microsoft is ending support for Windows 8.1 on January 10, 2023](https://www.microsoft.com/en-us/windows/end-of-support?r=1)

... so I don't think any of this is unreasonable.

Also in Microsoft Store: Windows 8.1 reaches End of Support on January 10, 2023. After this date, Windows 8.1 customers won't be able to purchase or acquire new content from the Store or through in-app purchases of apps and games running on Windows 8.1. As a developer, you can continue to submit updates of your products until June 30, 2023. Details are available from [www.microsoft.com/windows8](http://www.microsoft.com/windows8)

any reason why you still need to be on windows 7?

**Edit:** for people who end up here, you can disable auto update notifications inside settings

**Reply by uomoukko** [reply]
at least you could provide two builds, one with the old Electron 22, one with the new Electron 23. This way users with windows 7,8,8.1 could run this software with the bugfixes and upgrades. What do you think?

**Reply by mifi** [reply]
Theoretically yes, but the problem is that eventually the newest version of electron and 22 are going to drift apart in their apis. Then eventually the electron 22 version of losslesscut will break

**Reply by gnevan** [reply]
I vote up for a parallel branch for older win versions while it is still possible. Can the code be reused?

**Reply by mifi** [reply]
Yes, you can fork losslesscut on github and make your own branch for windows 7

**Reply by gnevan** [reply]
Yes, sure I can, but for you it may be a lot easier and handier. Maybe you need money to maintain that branch? I inderstand it takes more time.

**Reply by mifi** [reply]
I need time and motivation to maintain a separate branch, because for every new feature added to losslesscut, it adding it in two branches, adds a lot of extra work

**Comment by Murosel** [comment]
Thank you for the prompt response.

I still have v3.50.00 working so all is not lost.

I do not want this thread to turn into an OS debate, so suffice to say that Windows 7 is all I need to be able to do the things I need to do. I am aware of the EoS but that just means no further official updates, not that Win 7 suddenly stops working ;)

Thanks again.

**Reply by JeanPaulLucien** [reply]
Just use other software instead Electon's.

**Comment by Gitoffthelawn** [comment]
I, too, do not want to get into some tangential OS debate.

I hope this project can support Windows 7/8/8.1 as long as possible.

Windows 7/8/8.1 still have many real-world uses:

1. On machines isolated from the internet (including dedicated video processing boxes).
2. For non-profits organizations that do not have sufficient income streams to afford new hardware/software.
3. On systems that use software that is incompatible with Windows 10/11.
4. On systems that have hardware that is incompatible with Windows 10/11, especially those used by the disabled and lower-income persons.
5. On machines supported outside of Microsoft.  There are some incredibly talented companies supporting (and patching) Windows 7, such as [0Patch](https://0patch.com). Unlike Microsoft patches, 0Patch patches actually work and do not have a long and painful history of breaking existing functionality.

Furthermore, outside of the for-profit corporate world, there is no reliable method to prevent Windows 10 (which is nearing "End of Life", itself) from sending unknown data to closed-sourced servers whose owners cannot be verified.  To the best of my knowledge, Microsoft *still* has not disclosed any way for people to verify what data Windows 10 (and I think Windows 11) is sending to external servers (or better yet, a reliable method to prevent all data from being sent in the first place).  As such, non-corporate installations of Windows beyond Windows 8.1 are literally leaking unknown data to unverifiable servers.

**Reply by aberfan** [reply]
> Windows 7/8/8.1 still have many real-world uses:

Thank you for posting everything I would have typed.

I've come to this thread late, looking to see what the current LosslessCut was. I don't use it often (obviously) but when I do, I edit a very large amount of videos.

It's so disappointing to see projects exclude users.  After all, Windows 10/11 users can run Windows 7 apps, so why not allow the broadest user base, instead of enforcing Microsoft's business model?  Windows 7 still works brilliantly and stably for my usage; what amazing things does Electron 23 provide that made it necessary, I wonder, to move to it?

Anyway, has anyone done a fork yet, or found another program like this, but where perhaps the developer isn't so scornful of people who want to use their work?

**Reply by glodomorrapl** [reply]
if you think about it, people using an outdated operating system with estimated 2.6% desktop market share asking for some open-source project to support their ancient os is silly. if you can use a 2009 operating system that ended support in 2020, you sure as hell can use a 2 year old version of such a program.

like, get real for one second and realize the position you're in. please.

**Reply by aberfan** [reply]
If one makes decisions based on 'market share', then sure.  It's not a request so much to _support_ an ancient OS, as ask not to _unsupport_ it.  Are there features that require updated Electron?  It becomes a thing of "it's new so it must be better', but it rarely is.

There's so much harping on this "support ended", which is a perspective I have a hard time understanding. My car's old, and works. Why would I replace it? For that matter, my pants are old, but fit. The chair I'm sitting in, too, is quite old, but still functions.   I know many people like to constantly buy new things because they think they need to, but there are also those of us, who are content, when we find something that works, to continue using what works.

Windows 7, like Windows XP before it, is stable and fast.  Windows XP was good enough to woo me from Linux back in '01.  My Windows 7 system's uptime's over a year.  So what if it's old and 'unsupported'?  What am I missing out on? Microsoft's latest "AI" snooping that is a huge privacy violation?

Sorry, this is turning into a bit of a rant, so please accept that there are many people out there who choose to run an ancient OS that's unsupported for many reasons that feel valid to them.

Of course open source developers needn't cater to anyone at all; there's no entitlement or expectation.

However, the complaint is valid, that if you're taking the time to develop something and share it, presumably so people can enjoy it, then it's confounding to exclude groups of people, no matter how marginal, without good reason.    If it were something like LosslessCut needing GPU apis only present in Windows 10, that did magical stuff, that's one thing. But, "Oh, this uses Electron, and Electon doesn't support this, so now we don't"... that's not a feature.  It just makes it so those of us who found this great and handy program, that allows quickly fixing up files, can't continue to benefit from any fixes.

And you know, by this logic of "there's only 2.6% market share" --- you could apply that to having handicap stalls in bathrooms, and stores carrying non-dairy milk, and all the other things that a very small subset of the population needs.

tl;dr excluding people unnecessary is bad.  Unless Microsoft's giving a kickback... why help them?

I'll end this by saying I posted in the first place because of how much I've enjoyed LosslessCut.  It's done exactly what I needed, and I'm disappointed and sad that I need to find something else now.

**Reply by mifi** [reply]
It’s not my call to «unsupport» windows 7. it’s google’s (chromium). and yes electron needs to be kept up to date for many reasons including security updates. And yes losslesscut uses apis that are no longer available in windows 7, through electron. Still using windows 7 is a choice. You could use windows 11 or if you don’t want to purchase a new computer, you can use linux instead if you wanted to (which supports older hardware) but you choose to use windows 7 instead which is no longer updated. Handicap is not a choice. Also on the contrary I think the car analogy proves the exact opposite because you can still keep using old versions of losslesscut, just like you can keep using your old car, but the old car won’t get any improvements (unless you improve it yourself, which you can still do with losslesscut because it’s open source but it requires effort from your part.)

**Reply by glodomorrapl** [reply]
to add onto the car analogy, it would be accurate if your car was also constantly connected to the ever evolving piece of technology with several dozens of separate pieces of software each getting fresh vulnerabilities discovered on a daily basis (the internet). though this might eventually become accurate with the cars of today, several years into the future...

and yeah, mifi's got it. you can drive a 20 year old sh*tbox legally (in most places at least), it'll get the work done, but don't expect to be able to install tesla's autopilot on it. same with old operating systems. keep your expectations in check, and don't write paragraphs online because a modern program chooses not to support an abandoned OS.

**Comment by mifi** [comment]
i also dont want an OS war but here goes:

1. I think Windows 10 can also run isolated from the internet too? Just don’t connect it to the internet, or use a router that blocks all except certain traffic.

For the other points, I think Linux could be an alternative, as it is free and open source and it’s a good thing to motivate more people to  move away from closed source OS like windows. Or if not feasible, then run LosslessCut in Virtualized linux inside VirtualBox for example from the legacy Windows OS. LosslessCut doesn't use much CPU so I think running it in a VM can give very good performance too.

If you don’t trust a corporation to not send your data off to unknown servers, then Linux should be a much better alternative because you have full control and better privacy. And you can run Windows 7,8,9,10 inside virtualbox on linux and selectively block its access to the network

**Comment by Gitoffthelawn** [comment]
@mifi Thanks for the positive "no war" agreement/stance/maturity, and your constructive post.  As soon as I posted my words, I immediately felt that "internet regret" feeling of "oh brother, how tiresome are the responses to this going to be?".  Your post helped reaffirm my faith in the goodness and decency of people. :)

I agree that Linux is the best long-term solution for most situations.  I actually had written a paragraph about that, but felt it made my post too long, so I removed them before posting. :)

Short-term, however, I have trouble convincing everyone to switch to Linux.  The primary inhibitors tend to be:
1. Many people (especially elderly) are familiar with Windows. There are Linux distros that mimic the look and feel of Windows, but once you recommend someone change their OS, it can be an uphill battle.
2. Hardware support. Personally, I've never had a problem with hardware support on Linux (well, not in the last decade).  But many people still claim it's an issue.
3. The "if it ain't broke, don't fix it" philosophy.
4. Software availability. Often people have 1 or 2 Windows applications that they use all the time that have no Windows version.  WINE has improved tremendously over the years, but some applications still don't work well with it.  Installing Windows inside virtualbox on Linux can sometimes be an option (but may require purchasing more Windows licenses), but unfortunately that can be quite a bit of cost, effort, and maintenance to run one or two apps (although the effort goes down as one learns to get more efficient with the process).
5. Managers that refuse (they don't want anyone spending time on anything "nonessential").

That said, I think Linux is truly the most appropriate OS for most situations.  If only Linux had a marketing budget like Apple and Microsoft.

You mentioned:
> I think Windows 10 can also run isolated from the internet too? Just don’t connect it to the internet, or use a router that blocks all except certain traffic.

My experience (which is just my experience) is that this, unfortunately, is not really possible.  Microsoft has locked down Windows 10 (and 11, to my knowledge) to the point where they require internet access to work.  And because Microsoft refuses (to my knowledge, and after extensive research and multiple formal written requests) to fully disclose how to decipher what data they are sending to external servers, it is impossible to block only the desired packets or to even confirm what data they are truly sending from your system.

Outside of corporate use, it astonishes me that people voluntarily use Windows 10 and 11.  I think it's that people are basically lazy and/or overwhelmed by life, and that so many computers come with Windows 10/11 (due to aggressive bundling contracts by Microsoft and hardware manufacturers which have come under much, but insufficient, scrutiny) and it's easier for them to just stick with Windows 10/11 than switch to Linux (or even Windows 7/8/8.1).  It's the same reason that so many people still use the same short password on all their accounts, and that `123456` is still the most commonly used password in the world.

**Reply by mifi** [reply]
these are all valid arguments. I'm very surprised that windows 10/11 will stop working if disocnnected from the internet though! I wonder if it's because microsoft wants to prevent businesses from using windows for free/cheap in specific applications like kiosk applications, industrial computers etc that don't require internet access.. maybe they want people to instead buy a more expensive license that can work offline

**Reply by glodomorrapl** [reply]
about the last paragraph - sure, you could grab a brand new computer and install Windows 7/8 on it, but it won't work properly. since companies don't bother releasing drivers for EoL systems, performance and power usage will be worse (mostly related to CPU scheduler and missing GPU drivers) and some functionality might be broken (like USB 3 ports falling back to 2 or even not working at all, no wifi/ethernet etc). so if anyone dislikes Windows 10/11 and can't deal with linux (and let's not fool ourselves, most people can't), then yeah, they're stuck with these two Windows systems.

**Comment by DOSforever** [comment]
那么，最晚支持 Windows 7 的是 3.50 版还是 3.51.1 版？为什么 3.51.1 版只有源码下载？
So, is version 3.50 or 3.51.1 the latest to support Windows 7? Why is version 3.51.1 only available as a source code download?

**Reply by mifi** [reply]
I deleted [3.51.0](https://github.com/mifi/lossless-cut/releases/tag/v3.51.0) and [3.51.1](https://github.com/mifi/lossless-cut/releases/tag/v3.51.1) downloads because they had critical bugs

**Comment by stamminator** [comment]
Not sure if it's still relevant, but Electron 22 support was extended to [October 10, 2023](https://www.electronjs.org/blog/electron-22-0#windows-7881-support-update).

**Comment by hellboy1123** [comment]
> While 3.50.0 works fine on Windows 7, v3.52.0 does not; it reports an error of cannot find GetPackageFamilyName entry point in Kernel32.dll (works fine on Windows 10)
> 
> It seems odd to drop Windows 7 support in a minor update so I suspect this is either a problem with update, or I am missing a Windows Update at my end.
> 
> Any clues would be appreciated,
> 
> Thanks

Hi there,

It appears that you're experiencing an issue with version 3.52.0 of a certain software on Windows 7. While version 3.50.0 works fine on Windows 7, the newer version throws an error related to the "GetPackageFamilyName" entry point in Kernel32.dll. This discrepancy is indeed unusual since dropping Windows 7 support in a minor update is unexpected. It's possible that the problem lies either with the update itself or with a missing Windows Update on your end.

To troubleshoot this issue, here are a few suggestions:

Ensure Windows Updates are up to date: Verify that your Windows 7 operating system has all the necessary updates installed. Sometimes, missing updates can cause compatibility issues with certain software versions.

Check for software updates: Make sure you have the latest version of the software installed. Check the software  or [app](https://stickwarapk.download/) developer's website or official channels for any updates or patches specifically addressing Windows 7 compatibility.

Contact the software developer: Reach out to the software developer's support team or community forums to report the issue. They may be aware of the problem and can provide you with specific troubleshooting steps or a workaround.

Consider upgrading to a newer operating system: Windows 7 reached its end of life in January 2020, which means it no longer receives regular security updates from Microsoft. It might be worth considering an upgrade to a newer version of Windows that is still actively supported, such as Windows 10, to ensure better compatibility with the latest software releases.

**Comment by i486** [comment]
If anyone stumbled upon this discussion while searching for solutions, try [VXKex](https://github.com/i486/VxKex) a shot. LosslessCut latest version (3.59.1 at the time of writing) appears to be working fine on Windows 7 with VxKex.

**Reply by DOSforever** [reply]
Ha ha! Thumbs up for your username! :+1: I love the DOS era! :smile: :kissing_heart:

**Comment by mattack1** [comment]
> On Nov 30, 2024, at 6:34 AM, Dex ***@***.***> wrote:
> I'll end this by saying I posted in the first place because of how much I've enjoyed LosslessCut. It's done exactly what I needed, and I'm disappointed and sad that I need to find something else now.
> 

but you don't need to.. you can keep using whichever version works for you, with the same features it had the day you started using it..

(i.e. including whatever bugs that version had...)
